File: day1/day1.js
var fs = require('fs');

console.log("Santa ended on floor " + santasFloor('input.txt'));
console.log("Santa Entered the Basement on move " + santasInTheBasement('input.txt'));

function santasFloor(file) {
  var data = fs.readFileSync(file, 'utf-8').trimRight().split("");

  return move(data);
}

function santasInTheBasement(file) {
  var data = fs.readFileSync(file, 'utf-8').trimRight().split("");

  return move(data, true);
}

function move(data, check) {
  var floor = 0;
  var lastFloor;

  data.forEach(function(move, index) {
    if (!lastFloor) {
      if (check && floor === -1) {
        lastFloor = index;
      } else if (move === "\(") {
        floor += 1;
      } else if (move === "\)") {
        floor -= 1;
      }
    }
  });

  return lastFloor || floor;
}

File: day1/day1.rb
floor = 0
inputs = File.read('input.txt').rstrip.split("")
basement_reached = false;
moves_until_basement = false;

inputs.each_with_index do |input, index|
  if input == "("
    floor += 1
  elsif input == ")"
    floor -= 1
    if floor == -1 && !basement_reached
      moves_until_basement = index + 1
      basement_reached = true
    end
  end
end

puts "Part 1: #{floor}"
puts "Part 2: #{moves_until_basement}"
